main.py

```diff
@@ -0,0 +1,546 @@
+# The entrypoint for VxIngest
+# Responsible for checking the existing jobs documents and determining which need to be run
+
+import argparse
+import logging
+import os
+import shutil
+import sys
+import tarfile
+from datetime import datetime, timedelta
+from multiprocessing import Queue
+from pathlib import Path
+from typing import Callable, Optional, TypedDict
+
+import yaml
+from couchbase.auth import PasswordAuthenticator  # type: ignore
+from couchbase.cluster import Cluster  # type: ignore
+from couchbase.exceptions import CouchbaseException, TimeoutException  # type: ignore
+from couchbase.options import (  # type: ignore
+    ClusterOptions,
+    ClusterTimeoutOptions,
+    QueryOptions,
+)
+from prometheus_client import CollectorRegistry, Counter, Gauge, write_to_textfile
+
+import ctc_to_cb.run_ingest_threads
+import grib2_to_cb.run_ingest_threads
+import log_config
+import netcdf_to_cb.run_ingest_threads
+import partial_sums_to_cb.run_ingest_threads
+
+# Get a logger with this module's name to help with debugging
+logger = logging.getLogger(__name__)
+
+# Configure prometheus metrics
+# Note - we may need to import prometheus's multiprocessing libraries
+
+# Create a registry we can write to a file
+prom_registry = CollectorRegistry()
+
+# Use a gauge because we're doing 1 file per job run, a histogram could be more appropriate.
+# Note - if we used a historgram or summary, we could apply a decorator directly to the function we're interested in
+prom_duration = Gauge(
+    "run_ingest_duration",
+    "The duration of an ingest run, in seconds",
+    registry=prom_registry,
+)
+prom_successes = Counter(
+    "run_ingest_success_count",
+    "The number of successful ingest jobs",
+    registry=prom_registry,
+)
+prom_failures = Counter(
+    "run_ingest_failure_count",
+    "The number of failed ingest jobs",
+    registry=prom_registry,
+)
+prom_last_success = Gauge(
+    "job_last_success_unixtime",
+    "Last time a batch job successfully finished",
+    registry=prom_registry,
+)
+
+
+def process_cli():
+    """Processes the following flags
+
+    -c - credentials file path
```

Comment:
-j -s -e (and we need a -f) are all optional. -f is for file_pattern which is optional but needed for the grib and netcdf builders. We could also use the -t for threads. I think I disabled that actual parameter but we should put it back.

Reply:
That's good to know, thanks! I'll go ahead and add `-f/--file_pattern`, and `-t` for threads.